CONTRIBUTING.md needs updating

- [x] npm install --only=dev fails in root of repo with
    ```bash
    npm WARN saveError ENOENT: no such file or directory, open '/mnt/c/repos/azure-kusto-node/package.json'
    ```
- [x] npm install --only=dev for azure-kusto-data fails with
    ```
    source/connectionBuilder.ts:4:36 - error TS2307: Cannot find module '@azure/msal-common' or its corresponding type declarations.
    ```
- [x] npm install --only=dev for azure-kusto-ingest fails with
    ```
    source/descriptors.ts:6:30 - error TS2307: Cannot find module 'uuid' or its corresponding type declarations.
    ```

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -20,17 +20,42 @@ These include testing related packages as well as styling ([eslint](https://esli
 
 This project uses [mocha](https://mochajs.org/) for testing.
 
+First, enter the directory of the package to run the tests on:
+```bash
+cd azure-kusto-data
+# or:
+cd azure-kusto-ingest
+```
+
 In order to run unittests execute:
 
 ```bash
 npm test 
 ```
 
+Tests that need credentials to run will not run by default. 
+To run them, set the enviornment variable `LOGIN_TEST=1`
+These tests will usually need more credentials, which will be passed via enviornment variables.
+See tests that are declared with `loginTest` or `manualLoginTest` for their requirements.
+
 In order to check coding style compliance excute:
 
 ```bash
 npm run lint
 npm run checkFormat
+```
+
+To run end to end tests from the `azure-kusto-ingest` packge, you will need to set up environment variables:
+```bash
+# The following instructions are for bash. When using a different shell, use the syntax for setting environment variables.
+export ENGINE_CONNECTION_STRING=https://<your-cluster>.kusto.windows.net
+export DM_CONNECTION_STRING=https://ingest-<your-cluster>.kusto.windows.net
+export TEST_DATABASE=<your-database>
+export APP_ID=<your-app-id>
+export APP_KEY=<your-app-key>
+export TENANT_ID=<your-tenant-id>
+
+npm run e2e
 ```
 
 On creating PR or pushing a commit there is an automated validation which runs lint, unittests and end2end tests.

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -5,24 +5,35 @@ follow the instructions provided in [Microsoft Azure Projects Contribution Guide
 
 ## Requirements
 
-In order to work on this project, we recommend using the dev requirements:
+In order to work on this project, we recommend using the both production and dev dependencies:
 
 ```bash
-npm install --only=dev
+cd azure-kusto-data
+npm install
+cd ../azure-kusto-ingest
+npm install
 ```
 
-These including testing related packages as well as styling ([eslint](https://eslint.org/))
+These include testing related packages as well as styling ([eslint](https://eslint.org/))
 
 ## Testing
 
 This project uses [mocha](https://mochajs.org/) for testing.
-In order to run UT execute:
+
+In order to run unittests execute:
 
 ```bash
 npm test 
 ```
 
-On creating PR or pushing a commit there is an automate validation which run lint, UT and E2E test.
+In order to check coding style compliance excute:
+
+```bash
+npm run lint
+npm run checkFormat
+```
+
+On creating PR or pushing a commit there is an automated validation which runs lint, unittests and end2end tests.
 
 ## PRs
 We welcome contributions. In order to make the PR process efficient, please follow the below checklist:
@@ -32,7 +43,7 @@ We welcome contributions. In order to make the PR process efficient, please foll
 * **PR comment explains the changes done** - (This should be a TL;DR; as the rest of it should be documented in the related issue).
 * **PR is concise** - try and avoid make drastic changes in a single PR. Split it into multiple changes if possible. If you feel a major change is needed, it is ok, but make sure commit history is clear and one of the maintainers can comfortably review both the code and the logic behind the change. 
 * **Please provide any related information needed to understand the change** - docs, guidelines, use-case, best practices and so on. Opinions are accepted, but have to be backed up.
-* **Checks should pass** - these including linting with black and running tests.
+* **Checks should pass** - these include running coding style compliance check and unittests noted above.
 
 ## Code of Conduct
 This project's code of conduct can be found in the

